tensorflow_datasets/core/lazy_imports_lib.py

```diff
@@ -20,24 +20,98 @@
 from __future__ import division
 from __future__ import print_function
 
+import contextlib
 import importlib
+import sys
+import types
 
 from tensorflow_datasets.core.utils import py_utils as utils
 
+_ERR_MSG = ("Failed importing {name}. This likely means that the dataset "
+            "requires additional dependencies that have to be "
+            "manually installed (usually with `pip install {name}`). See "
+            "setup.py extras_require.")
+
+# `lazy_imports` currently only allow the following external dependencies to
+# be used during dataset generation.
+_ALLOWED_LAZY_DEPS = [
+    "apache_beam",
+    "crepe",
+    "cv2",
+    "h5py",
+    "langdetect",
+    "librosa",
+    "matplotlib",
+    "matplotlib.pyplot",
+    "mwparserfromhell",
+    "nltk",
+    "pandas",
+    "PIL_Image",
+    "PIL.TiffImagePlugin",
+    "pretty_midi",
+    "pydub",
+    "scipy",
+    "scipy.io",
+    "scipy.ndimage",
+    "skimage",
+    "skimage.color",
+    "skimage.filters",
+    "skimage.external.tifffile",
+    "tensorflow_io",
+    "tldextract",
+    "os",
+    # "test_foo",
+]
+
 
 def _try_import(module_name):
   """Try importing a module, with an informative error message on failure."""
   try:
     mod = importlib.import_module(module_name)
     return mod
   except ImportError:
-    err_msg = ("Failed importing {name}. This likely means that the dataset "
-               "requires additional dependencies that have to be "
-               "manually installed (usually with `pip install {name}`). See "
-               "setup.py extras_require.").format(name=module_name)
+    err_msg = _ERR_MSG.format(name=module_name)
     utils.reraise(suffix=err_msg)
 
 
+class FakeModule(types.ModuleType):
+  """A fake module which raise ImportError whenever an unknown attribute is accessed"""
+  def __init__(self, name):
+    self.__path__ = None
```

Comment:
Yeah, we don't need to change the value of `__path__`. Sorry, about that.

`__path__` variable was accessed by some `importlib` module functions. So I had to create it to prevent `__getattr__` from throwing any errors.

Reply:
I was surprised to discover that ModuleType does not have `__path__`.
Is there a better parent `FakeModule` can inherit from?